docs/core/entity/climate.md

```diff
@@ -139,6 +139,48 @@ class MyClimateEntity(ClimateEntity):
         """Set new target hvac mode."""
 ```
 
+### Turn on
```

Comment:
We should explain that the expected outcome of a successful `climate.turn_on` call is that the hvac_mode is different from `HVACMode.OFF`, either optimistically set by the service handler or with the next state update.